src/Collection/Collection.model.ts

```diff
@@ -19,11 +19,31 @@ export class Collection extends Model<CollectionAttributes> {
       WHERE id = ANY(${ids})`)
   }
 
-  static async isValidName(id: string, name: string) {
+  // TODO: should we have unique names for all TPC collections?
+  /**
+   * Checks if a collection name is valid.
+   * A collection name is valid if:
+   * - There's no other collection that has the given name other than the one with the given id.
+   * - There's no other collection that has the given name other than the one with the given id and urn suffix.
+   *
+   * @param id - The collection id.
+   * @param name - The collection name to be changed or created.
+   * @param urn_suffix - The collection URN suffix if the collection is a TWP one.
+   */
+  static async isValidName(
+    id: string,
+    name: string,
+    urn_suffix?: string
+  ): Promise<boolean> {
     const counts = await this.query(SQL`
     SELECT count(*) as count
       FROM ${raw(this.tableName)}
       WHERE id != ${id}
+        AND (urn_suffix IS NULL${
+          urn_suffix
+            ? ` OR LOWER(urn_suffix) = ${urn_suffix.toLowerCase()}`
+            : ''
+        })
```

Comment:
This shouldn't really work! the `SQL` function takes a template, grabs each interpolated value and constructs a valid node-psql query like:

```ts
const id = 3
const name = 'hey'
const a = SQL`
  SELECT * FROM collections WHERE id = ${id} AND name = ${name}`

a // => { query: 'SELECT * FROM collections where id = $1', values: [3, 'hey'] }
```

To append arbitrary data we use `raw` (or `SQL.raw`), but we have to be careful no to do it with data we do not control. `this.tableName` is fine, `urn_suffix` might not be.

This query would then result in something like:

```sql
    SELECT count(*) as count
      FROM collections
      WHERE id != $1
        AND (urn_suffix IS NULL$2)
        AND LOWER(name) = $3
```

Breaking with ` syntax error at or near \"NULL$2\"`

You could instead append `SQL` functions, as they support that kind of behaviour:

```ts
    const a = SQL`
    SELECT count(*) as count
      FROM ${raw(this.tableName)}
      WHERE id != ${id}
        AND (urn_suffix IS NULL ${
          urn_suffix
            ? SQL`OR LOWER(urn_suffix) = ${urn_suffix.toLowerCase()}`
            : SQL`OR 1 = 1`
        })
        AND LOWER(name) = ${name.toLowerCase()}`

// Resulting in:
`SELECT count(*) as count
      FROM collections
      WHERE id != $1
        AND (urn_suffix IS NULL OR LOWER(urn_suffix) = $2)
        AND LOWER(name) = $3`
        
// or

`SELECT count(*) as count
      FROM collections
      WHERE id != $1
        AND (urn_suffix IS NULL OR 1 = 1)
        AND LOWER(name) = $2`
```